caveclient/skeletonservice.py

```diff
@@ -0,0 +1,116 @@
+
+from .auth import AuthClient
+from .base import (
+    ClientBase,
+    _api_endpoints,
+    handle_response,
+)
+from .endpoints import skeletonservice_api_versions, skeletonservice_common
+from typing import Optional
+
+SERVER_KEY = "skeleton_server_address"
+
+
+'''
+Usage
+'''
+
+class SkeletonClient(ClientBase):
+    def __init__(
+        self,
+        server_address,
+        auth_client=None,
+        api_version="latest",
+        verify=True,
+        max_retries=None,
+        pool_maxsize=None,
+        pool_block=None,
+        over_client=None,
+    ):
+        if auth_client is None:
+            auth_client = AuthClient()
+
+        auth_header = auth_client.request_header
+        endpoints, api_version = _api_endpoints(
+            api_version,
+            SERVER_KEY,
+            server_address,
+            skeletonservice_common,
+            skeletonservice_api_versions,
+            auth_header,
+            fallback_version=1,
+            verify=verify,
+        )
+
+        super(SkeletonClient, self).__init__(
+            server_address,
+            auth_header,
+            api_version,
+            endpoints,
+            SERVER_KEY,
+            max_retries=max_retries,
+            pool_maxsize=pool_maxsize,
+            pool_block=pool_block,
+            over_client=over_client,
+        )
+    
+    def get_skeleton_info(
+        self,
+        datastack_name,
+    ):
+        """Gets basic skeleton information for a datastack
+
+        Parameters
+        ----------
+        datastack_name : str
+            The name of the datastack to check
+
+        Returns
+        -------
+        bool
+            JSON description
+        """
+        endpoint_mapping = self.default_url_mapping
+        endpoint_mapping["datastack_name"] = datastack_name
+
+        url = self._endpoints["skeleton_info"].format_map(endpoint_mapping)
+
+        response = self.session.get(url)
+        return handle_response(response, False)
+    
+    def get_skeleton(
+        self,
+        root_id: int,
+        datastack_name:   Optional[str] =None,
+        skeleton_version: Optional[int] =0,
+        output_format:    Optional[str] ='precomputed',
```

Comment:
Is `output_format` nullable or one of several parameters? I think you might mean `Literal['precomputed',..other options..]` here.

Reply:
Plus, of course, since it has to be formatted into the URL, some value has to be provided for it. This is why I originally had multiple endpoints, but the recommendation was to strip it down to a single "get_skeleton", which means all URL parameters have to be supplied every time (what does it mean for a REST value to be an empty string, much less a None)?